### PR TITLE
Default stylesheets revision + clean-up

### PR DESCRIPTION
**Pull request by MisterMakerNL**
This PR included 2 new stylesheets they are intended to be the only 2 included in main. It is a dark and a light version.
Any other stylesheets should be included as an addon.
image
image

image
This also updated the look of Startpage 
image

### CODE REVIEW

**Comment by PaddleStroke** [comment]
Interesting. I seem to recall that Qt 6.5 was implementing a dark mode that would make things simpler. Is that taking advantage of that? Are we even using 6.5 btw?

**Comment by MisterMakerNL** [comment]
@PaddleStroke this would indeed work very well with the 6.5 functionality as I understand Qt 6.5 can detect the dark style of windows and apply a dark theme. But it doesn't have a good dark theme out of the box. 
But would be more logical, if it did so maybe they have added one. QT docs aren't really complete most of the time.

**Comment by MisterMakerNL** [comment]
Updated, already better than currently included stylesheets! So pretty much ready, just going to spend some time fixing up overlay.

image

**Comment by MisterMakerNL** [comment]
Btw if people think this is not modern or ugly because of the purple gradients, then we should fix that in no-stylesheet too and also here. This is a black version of no-stylesheet including the not so popular gradients. (although the menu gradients are adjusted to the white text.)

Might be a good idea to adjust the background color to the same gradient as in the menu like this:
image
I updated the Qsint group which now looks like original.
I also fixed the overlay a bit. worked around the properties panel click true issue by making the background 0.01% transparency.
image

https://github.com/FreeCAD/FreeCAD/issues/13351

**Comment by MisterMakerNL** [comment]
Added light theme 
image

**Comment by MisterMakerNL** [comment]
Added overlay fixes:
image

**Comment by MisterMakerNL** [comment]
It probably has some tiny bugs, I see one with the treeview having black arrows.
But it's still a lot better than current situation, so this can be reviewed

**Comment by chennes** [comment]
Are these meant to replace the stylesheets used in the "Dark" and "Light" preference packs?

**Comment by MisterMakerNL** [comment]
> Are these meant to replace the stylesheets used in the "Dark" and "Light" preference packs?

Yes, I'll have a look at making the preference packs too.

**Comment by chennes** [comment]
@FreeCAD/design-working-group can you please take a look?

**Comment by pierreporte** [comment]
@chennes Looking at the screenshots, they look pretty decent. The only problem is the background but it has already been discussed: needs #13980 to be fixed before. The themes may not be perfect but they are good enough IMO and they can always be improved.

**Comment by MisterMakerNL** [comment]
The idea here is also to work from this as a base.
Also move no-stylesheet to dev features.
These 2 styles and the OS-style. I see as the 3 options, in Windows you would have only 2 styles. 
If you pick OS-style it would just pick dark or light on the case how you have set your windows theme.
I think this how most programs work.

Then OpenThemes and Prodark and Behave-dark, pre-installed as an addon. (as long as they are actively maintained.)
I do not plan to maintain darker dark dork modern and such. It's to much work for what it's worth in my opinion. So I'd suggest to remove them unless someone wants to maintain them. 
 have a backup as an addon, which will have the last version as was included in Freecad.

**Comment by FEA-eng** [comment]
> in Windows you would have only 2 styles

The old classic (current default) and the new dark classic? Only those would be shown in the first start widget? Makes sense but maybe the preinstalled OpenThemes (which are great and IMO should be included in the default package as you said) should also be shown there to make it clear for all users that they are already available. They are so good that one of them (likely Light) could even become the new default style but it's not a must for me.

**Comment by MisterMakerNL** [comment]
I don't think Opentheme is ready, although the preference pack and the backgrounds are.
I have issues with the Stylesheets, as I also explained on Discord.
To sum it up:

1. The compile system that is used to compile the QSS files is in my view unnecessary complicated. And will make it harder for anyone. Especially non-linux users. Like you need git that is pretty obvious and easy for any Linux user, but anyone else would have no clue where to start.
2. The compile workflow can easily be broken by anyone who doesn't realizes it is compiles by editing the QSS files directly.
3. It still has a lot of tiny bugs that these stylesheets don't have. 

I am also aware that this compile system has benefits, but I really don't think we need those. It's not worth it. 

Also these Stylesheets are not that different from Opentheme.

Only a couple of things that really make these different from Opentheme are there too keep the Freecad look, but can easily be removed/changed.

1. Tabs
2. The use of the purplisch gradients.

**Comment by obelisk79** [comment]
Regarding the use of sass for the stylesheets. I don't think it is that confusing or complicated at all. This is something that is extremely common in the web development world where stylesheets are commonplace.  

I agree however regarding the little bugs, I've been chipping away at those. Some are rather problematic to isolate.

**Comment by kadet1090** [comment]
IMHO These styles should be merged only if the rest of preinstalled are removed. I however don't like that even though the stylesheets are aiming to be "simple", the source files are almost 3000 lines of code! It is a lot of code to maintain and to understand.

**Comment by FEA-eng** [comment]
@kadet1090 So you are against merging this now? What about the rest of the DWG? @maxwxyz?

**Comment by MisterMakerNL** [comment]
He doesn't say he is against merging it, he rather have all the others be removed. 
Which I agree on. But I already made a PR about that which wasn't successful 
https://github.com/FreeCAD/FreeCAD/pull/11982

**Comment by kadet1090** [comment]
> He doesn't say he is against merging it, he rather have all the others be removed.

Yes, this is closer to my opinion. I'm in favor of merging this as long as we will get rid of other pre-installed themes because if we don't do that we add next 6k lines of qss that must be maintained and virtually no-one does that resulting in overall poor quality of included themes. 

We should focus on having two well maintained ones in the core and not a dozen. As long as we don't get rid of other I'm against merging another ones as I fear that they will end up like the rest.

**Comment by PaddleStroke** [comment]
+1, we need 2 basic themes well maintained. The rest should be addon.

**Comment by FEA-eng** [comment]
> +1, we need 2 basic themes well maintained. The rest should be addon.

I would say 3 - classic (current default - for those who would heavily complained if it was removed) and 2 modern ones (light and dark).

**Comment by kadet1090** [comment]
Yes, I think of classic as "no theme"

**Comment by MisterMakerNL** [comment]
I did a lot of work today, also compiled it tested it.
Had no issues.
There is only a tiny overlay text color issues in the light theme.
PR is a bit big, although not that complicated as I only change code and didn't really added any.

File: src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png
Comment (maxwxyz) [inline src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png:1]: I have no idea about styles but is it wanted that they are tilted? Can we replace them with .svgs? I can help.

File: src/Mod/Start/Gui/Resources/thumbnails/Classic418.png
Comment (maxwxyz) [inline src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Can I provide new ones for these 3? This is all an example for a Don't use according to the stylesheet. Will classic still has the same BG color?

**Review by maxwxyz — CHANGES_REQUESTED** [review]

File: src/Mod/Start/Gui/Resources/thumbnails/Classic418.png
Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Off-course the idea of these they show the colors used in the theme, but in the Freecad logo.
I do think they are too big.
Not sure why @chennes added the dimensions in the filename but since he knows way more about this then me I did the same.

Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Theme selection

File: src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png
Comment (MisterMakerNL) [reply src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png:1]: No svg anymore those give scaling issues. PNG seem to render way nicer in QT especially on weird shaped icons.
These icons definitely shouldn't be slightly rotated. 
I'll fix that!
I do have a big svg file where these icons are in and can easily be exported. Not sure where I should upload this.

File: src/Mod/Start/Gui/Resources/thumbnails/Classic418.png
Comment (maxwxyz) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: no :) either full color icon or the mono one in white, blue or red on dark backgrounds.

Comment (sytabaresa) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: hey @MisterMakerNL great changes!
Only one recommendation, when you use mono version, make the F transparent, looks better:
image

Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: The colors are based on the colors of the stylesheet :)
So changing the colors will break the functionality!

Comment (maxwxyz) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Maybe we should not display colors with the icon but with e.g. a toolbar / task panel with the color. I was in favor of the schematic images you did.

Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: > Maybe we should not display colors with the icon but with e.g. a toolbar / task panel with the color. I was in favor of the schematic images you did.

I thought showing off the new logo was way more fun!

Comment (maxwxyz) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: but it's now just 3 logos which have nothing to do with the theme unfortunately.
Maybe your proposal without the colorful icons, just the basic things as a demo of how it will look.
grafik

Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Something like this?

classic
dark
light

Comment (maxwxyz) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: I would remove the colorful icons in the top left. not sure about preselect and select color at the same time but it's definitely better than the logo version. But will the light background be light or does it stay almost the same as dark?

Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Light background is gonna be Light as soon as we fix the sketch lines color issue.

Comment (MisterMakerNL) [reply src/Mod/Start/Gui/Resources/thumbnails/Classic418.png:1]: Theme selection
This is the svg file if you want to have a go at it.

**Comment by MisterMakerNL** [comment]
> Do the default theme have control on the standard part color? The current one has too much contrast against the light background and not enough against the dark background.

I took all the preference-packs colors from OpenTheme, it seems that the light version isn't as far developed also the dark version. And needs some more work. 

I btw don't turn this pr into a draft because merging it would make Freecad still look better and will get me better/more bug reports.

**Comment by pierreporte** [comment]
Do you prefer keeping the color as is and have a separate issue opened after merging? Hopefully it can be fixed in time.

**Comment by MisterMakerNL** [comment]
> Do you prefer keeping the color as is and have a separate issue opened after merging? Hopefully it can be fixed in time.

I am trying to align the pr with the sketcher fixes from @kadet1090 . Hopefully then it is just tiny things.

**Comment by MisterMakerNL** [comment]
> Do you prefer keeping the color as is and have a separate issue opened after merging? Hopefully it can be fixed in time.

I am now so far that it's only small style bugs, bugs that you really have to encounter while using the program. And so I think it is wise to push this PR so I get more bug reports. Especially the switch to white background can cause some issues in workbenches.
@chennes Do you got some time too, this also changes the startpage a tiny bit.

**Comment by MisterMakerNL** [comment]
This also now changes the title bar in windows to black if you have dark mode on. Code thanks to @kadet1090 !

**Comment by pierreporte** [comment]
> I am now so far that it's only small style bugs, bugs that you really have to encounter while using the program. And so I think it is wise to push this PR so I get more bug reports. Especially the switch to white background can cause some issues in workbenches.

Understood. I’m waiting for merge then.

**Comment by chennes** [comment]
Could you please resolve the conflicts @MisterMakerNL?

**Comment by MisterMakerNL** [comment]
> Could you please resolve the conflicts @MisterMakerNL?

I think I fixed the issue on maingui.cpp the other merge issues are about files I removed not sure why these are issues and how to fix them.

**Comment by chennes** [comment]
I'm away from a computer for the weekend, but I can try to help get this cleaned up when I am back.

**Comment by PaddleStroke** [comment]
Youtuber mentioning the Theme issue : 
https://youtu.be/p8Pk1ayx6LQ?si=-EsV5F3azbJRI7Ld&t=119

**Comment by MisterMakerNL** [comment]
Dark looks like this 
image

File: src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png
Comment (MisterMakerNL) [reply src/Gui/Stylesheets/images_classic/Arrow-double-LeftRight-black.png:1]: I fixed this, although double arrows aren't used atm.

**Review by maxwxyz — APPROVED** [review]

**Comment by adrianinsaval** [comment]
any reason why "behave"  was kept around here? shouldn't we move it to an addon too?

**Comment by MisterMakerNL** [comment]
> any reason why "behave" was kept around here? shouldn't we move it to an addon too?

Because I tried removing it before and it didn't got approved since it's being maintained by @Syres916 . I didn't wanted to stop this PR for being merged, because we got into a discussion about this. I do understand why we should remove it, because it's confusing and the style doesn't fit with the others. I hope @Syres916 also realizes this and is willingly to move his theme to an addon.

**Pull request merged**
